client/src/components/productDetail/ProductRecommend/index.tsx

```diff
@@ -0,0 +1,39 @@
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
+
+import React, { FC } from 'react';
+import { useHistory } from '~/core/Router';
+import { ImageWrapper, ProductRecommendContainerWrapper } from './index.style';
+
+interface Props {
+  products: {
+    idx: number;
+    thumbnail: string;
+  }[];
+}
+
+const ProductRecommendContainer: FC<Props> = ({ products }) => {
+  const { push } = useHistory();
+
+  const handleImgClick = (idx: number) => {
+    push(`/products/${idx}`);
+  };
+
+  return (
+    <ProductRecommendContainerWrapper>
+      <div>추천상품</div>
+      <ImageWrapper>
+        {products.map((product) => (
+          <img
+            key={product.idx}
+            onClick={() => handleImgClick(product.idx)}
```

Comment:
useCallback으로 만들어서 바인딩 시키면 ProductRecommendContainer가 리렌더되더라도 해당 함수를 재생성하지 않을 것 같습니다!!

```suggestion
            onClick={handleImgClick}
```

Reply:
오홍 넵 수정할게요!!

Reply:
근데 해당 부분에서는 product의 idx를 넘겨주어야 해서

```
onClick={() => handleImgClick(product.idx)}
```

이렇게 할 수밖에 없지 않을까요?

onClick={handleImgClick(product.idx)}

요렇게하면 에러가,,